### PR TITLE
v3.0: PHP8, Callbacks, & BedrockEconomy

### PR DESCRIPTION
**Pull request by Ifera**

### CODE REVIEW

File: src/DaPigGuy/libPiggyEconomy/providers/EconomyProvider.php [hunk 1]
```diff
@@ -22,11 +22,31 @@ public function getMonetaryUnit(): string
...
+	 * If money was taken successfully then callable would contain true else false.
+	 */
+    abstract function takeMoney(Player $player, float $amount, ?callable $callback = null): void;
+
+	/**
+	 * $callback -> function(bool $success): void{}
+	 *
+	 * If money was set successfully then callable would contain true else false.
+	 */
+    abstract function setMoney(Player $player, float $amount, ?callable $callback = null): void;
```
Comment (Aericio) [inline src/DaPigGuy/libPiggyEconomy/providers/EconomyProvider.php:41]: @DaPigGuy dude get this he explains his code lol unlike someone

Comment (Aericio) [reply src/DaPigGuy/libPiggyEconomy/providers/EconomyProvider.php:41]: anyways need to reformat

Comment (Ifera) [reply src/DaPigGuy/libPiggyEconomy/providers/EconomyProvider.php:41]: kk

**Review by Ifera — COMMENTED** [review]
oh lol

**Review by DaPigGuy — APPROVED** [review]

**Pull request merged**